File: week_6/sentimental-readability/readability.py
from cs50 import get_string

# Get text sample
sample = get_string("Text: ")

letters = sents = words = 0

# Checks if it contains properly separated, non-symbol words
for ch in sample:
    if ch.isalpha():
        letters += 1
    if ch.isspace():
        words += 1
    if ch in ["?", ".", "!"]:
        sents += 1
words += 1

# Calculates each of the formula variables
L = (letters * 100.0) / words
S = (sents * 100.0) / words
formula = round(0.0588 * L - 0.296 * S - 15.8)

# Prints out the sample text Grade
if formula < 1:
    print("Before Grade 1")
elif formula >= 16:
    print("Grade 16+")
else:
    print(f"Grade {formula}")

File: week_2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <math.h>
#include <string.h>


int get_index(string sample);

int main(void)
{

    // Get text sample and print according to level
    string text = get_string("Text: ");
    int index = get_index(text);

    if (index < 1)
    {
        printf("Before Grade 1\n");
    }

    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }

    else
    {
        printf("Grade %d\n", index);
        return 0;
    }
}

int get_index(string sample)
{
    int letters = 0, words = 0, sents = 0;

    for (int i = 0; i < strlen(sample); i++)
    {
        // Count letters
        char ch = sample[i];
        if (isalpha(ch))
        {
            letters++;
        }

        // Count words
        if (isspace(ch))
        {
            words++;
        }

        // Count sentences
        if (ch == '!' || ch == '?' || ch == '.')
        {
            sents++;
        }
    }
    words++;

    // Plugging in the Coleman-Liau formula
    float L = (letters * 100.0f) / words;
    float S = (sents * 100.0f) / words;
    return round(0.0588 * L - 0.296 * S - 15.8);
}